src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -8737,6 +8803,605 @@ namespace SurfaceGeometry {
+		Real64 oppositeAzimuth = fmod(selectedAzimuth + 180., 360.);
+		Real64 selectedArea = Surface( selectedSurNum ).Area;
+		int selectedNumCorners = zonePoly.SurfaceFace( faceIndex ).NSides;
+		int found = -1;
+
+		for ( int iFace = 1; iFace <= zonePoly.NumSurfaceFaces; ++iFace ) {
+			int curSurfNum = zonePoly.SurfaceFace( iFace ).SurfNum;
+			if ( ( zonePoly.SurfaceFace( iFace ).NSides == selectedNumCorners ) && ( abs(Surface( curSurfNum ).Area - selectedArea) < 0.01) && ( abs( Surface( curSurfNum ).Azimuth - oppositeAzimuth ) < 1. ) ){
+				found = iFace;
+				break;
+			}
+		}
+		return found;
+	}
+
+	// tests if the corners of one face of the polyhedron are the same distance from corners of another face
+	bool
+	areCornersEquidistant(
+		DataVectorTypes::Polyhedron const & zonePoly,
+		int const & faceIndex,
+		int const & opFaceIndex,
+		Real64 & distanceBetween
+	)
+	{
+		// J. Glazer - March 2017
+
+		Real64 tol = 0.0254; //  2.54 cm = 1 inch 
+		bool allAreEquidistant = true;
+		Real64 firstDistance = -99.;
+		if ( zonePoly.SurfaceFace( faceIndex ).NSides == zonePoly.SurfaceFace( opFaceIndex ).NSides ) { // double check that the number of sides match
+			for ( int iVertex = 1; iVertex <= zonePoly.SurfaceFace( faceIndex ).NSides; ++iVertex ) {
+				int iVertexOpp = 1 + zonePoly.SurfaceFace( faceIndex ).NSides - iVertex; // count backwards for opposite face
+				Real64 curDistBetwCorners =  distance( zonePoly.SurfaceFace(faceIndex).FacePoints( iVertex ), zonePoly.SurfaceFace( opFaceIndex ).FacePoints( iVertexOpp ) );
+				if ( iVertex == 1 ) {
+					firstDistance = curDistBetwCorners;
+				} else {
+					if ( abs( curDistBetwCorners - firstDistance ) > tol ) {
+						allAreEquidistant = false;
+						break; 
+					}
+				}
+			}
+		} else {
+			allAreEquidistant = false;
+		}
+		if ( allAreEquidistant ) distanceBetween = firstDistance;
+		return allAreEquidistant;
+	}
+
+	// test if two points in space are in the same position based on a small tolerance
+	bool 
+	isAlmostEqual3dPt(
+		DataVectorTypes::Vector v1,
+		DataVectorTypes::Vector v2
+	)
+	{
+		// J. Glazer - March 2017
+
+		Real64 tol = 0.0127; //  1.27 cm = 1/2 inch 
+		return ( ( abs( v1.x - v2.x ) < tol ) && ( abs( v1.y - v2.y ) < tol ) && ( abs( v1.z - v2.z ) < tol ) ) ;
+	}
+
+	// test if two points on a plane are in the same position based on a small tolerance
+	bool
+	isAlmostEqual2dPt(
+		DataVectorTypes::Vector_2d v1,
+		DataVectorTypes::Vector_2d v2
+	)
+	{
+		// J. Glazer - March 2017
+
+		Real64 tol = 0.0254; //  2.54 cm = 1 inch 
+		return ( ( abs( v1.x - v2.x ) < tol ) && ( abs( v1.y - v2.y ) < tol ) );
+	}
+
+	// test if two points on a plane are in the same position based on a small tolerance (based on Vector2dCount comparison)
+	bool
+		isAlmostEqual2dPt(
+			DataVectorTypes::Vector2dCount v1,
+			DataVectorTypes::Vector2dCount v2
+		)
+	{
+		// J. Glazer - March 2017
+
+		Real64 tol = 0.0254; //  2.54 cm = 1 inch 
```

Comment:
They were both 1inch and had to fix the 3d function to 0.5 inch because it was being used and failing on a wall that was supposed to be 1 inch tall (a weird edge to an attic). I should go back and change the 2d compare to 0.5 inch also